zap/src/main/java/org/parosproxy/paros/network/HttpMessage.java

```diff
@@ -1090,4 +1094,22 @@ public void setResponseFromTargetHost(final boolean responseFromTargetHost) {
     public String getType() {
         return MESSAGE_TYPE;
     }
+
+    public static void replaceCsrfToken(
```

Comment:
- This would be better in `ExtensionAntiCSRF`, this class does not need to know about the anti-CSRF tokens.
 - It can be extracted the whole method (`regenerateAntiCsrfToken`) and also some of the previous logic (starting at `List<AntiCsrfToken> tokens = extAntiCSRF.getTokens(message);`) it's the same for both/all.

Reply:
> * It can be extracted the whole method (`regenerateAntiCsrfToken`) and also some of the previous logic (starting at `List<AntiCsrfToken> tokens = extAntiCSRF.getTokens(message);`) it's the same for both/all.

Did you mean that instead of just extracting `replaceCsrfToken` I could extract the entire method `regenerateAntiCsrfToken`?

Reply:
Yes.

Reply:
Just a note, I ran into some issues doing this because `AbstractPlugin` had it's own implementation of `sendAndReceive()` so I implemented a similar logic in the new extracted method

Reply:
It's better to leave the actual send to the caller, the `HttpSender` might have state (e.g. user, session) which would be lost if the message is sent by the extension. The method could accept a function which sends the `HttpMessage`, e.g.:
```Java
public interface HttpMessageSender {
    void sendMessage(HttpMessage message) throws IOException;
}
```

Reply:
I did not know that was possible haha, thanks again for pointing me in the right direction :)